db/view/view.cc

```diff
@@ -142,22 +142,63 @@ void view_info::set_base_info(db::view::base_info_ptr base_info) {
     _base_info = std::move(base_info);
 }
 
+// A constructor for a read and write enabled base info
+db::view::base_dependent_view_info::base_dependent_view_info(schema_ptr base_schema, std::vector<column_id>&& base_non_pk_columns_in_view_pk)
+        : base_schema{std::move(base_schema)}
+        , _base_non_pk_columns_in_view_pk{std::move(base_non_pk_columns_in_view_pk)}
+        , has_base_non_pk_columns_in_view_pk{!_base_non_pk_columns_in_view_pk.empty()}
+        , base_info_is_for_read_only{false} {
+
+}
+    // A constructor for a read only base info
+db::view::base_dependent_view_info::base_dependent_view_info(schema_ptr base_schema, bool has_base_non_pk_columns_in_view_pk)
```

Comment:
For reads, so we actually need base_schema? I think we don't - why would a read of the view table need the base schema? But maybe I'm missing something?
If we don't need the base_schema, we shouldn't pass it in....

Reply:
I don't think we need it for reads but I was too afraid to get rid of it, I think it was intended to prevent from the base schema object to be destroyed. I will do a pass on the code to see if we need it and if not I will get rid of it.

Reply:
After validating that we don't need it for reads I decided to take the same approach as with the pk column list, 
and have an accessor to the base schema that will assert if the view_info can only be for read only.
This will be in the next version I push.